### PR TITLE
[Saving Login Information] Storing login information in local storage

### PR DESCRIPTION
**Pull request by fnormand01**
# Description

This PR is introducing the storage of the login information into local storage, in order to keep the user logged in during his experience in our application. The token stays alive for 1 hour. After that, the concerned token will expired and the user will have to login again in order to re-access the dashboard page.

**A future PR will introduce a token refresh functionality, so that the users don't get logged out if they are using the app, even after 1 hour.**

This PR is also introducing some new urls. The new **home page** is now: 
- http://localhost/login

Once logged in, the home page becomes:
- http://localhost/dashboard

Also, as soon as the token expires, the user will be automatically redirected to the login page on his next action.

**Note that you don't need to be logged in to access the sample test.**

## Type of change

- New feature (non-breaking change which adds functionality)
- Code cleanliness or refactor

## Screenshot

**http://localhost/login**

image

**http://localhost/dashboard (you need to be logged in to see this page)**

image

**From now one, you can refresh the page, close your browser, change tabs, do whatever you want and you will stay logged in. After an hour, if you did not click _Logout_, the token will expire and you'll be automatically redirected to the login page.**

# Testing

Manual steps to reproduce this functionality:

1.  Don't forget that the home url is now: http://localhost/login
2.  You'll first need to change the JWT token expiration settings to test this new functionality by changing the current ones by the following:

image

3.  After that, login to the system, refresh the page, change tab, close the browser if you want and make sure that after 10 seconds, you get logged out automatically and redirected to the login page.

**Note that if the token expires when you are in the sample test, you will not be redirected to the login page, since you don't need to be authenticated in order to access this sample test. You will only be redirected if you are in your _dashboard_.**

# Checklist

Applicable for all code changes.

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new compiler warnings
- [ ] ~~My changes generate no new accessibility errors and/or warnings~~
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] ~~I have researched WCAG2.1 accessibility standards and met them in this PR (can be navigated using a keyboard)~~
- [x] My changes look good on IE 11+ and Chrome

### CODE REVIEW

File: frontend/src/modules/LoginRedux.js [hunk 1]
```diff
@@ -8,8 +10,22 @@ export const IS_CHANGING_PASSWORD = "IS_CHANGING_PASSWORD";
 export const CHANGE_PASSWORD_SUCCESS = "CHANGE_PASSWORD_SUCCESS";
 export const CHANGE_PASSWORD_FAILURE = "CHANGE_PASSWORD_FAILURE";
 
-// Simplified version of the authentication action (temporary)
-const authenticateAction = authenticated => ({ type: AUTHENTICATED, authenticated });
+function authenticateAction(response, dispatch, location, push) {
+  return async function() {
+    if (navigator.cookieEnabled) {
+      localStorage.setItem("token", response.token);
```
Comment (caleybrock) [inline frontend/src/modules/LoginRedux.js]: why did you rename it from "auth_token"? that seemed more explicit.

File: frontend/src/components/authentication/history.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { createBrowserHistory } from "history";
```
Comment (caleybrock) [inline frontend/src/components/authentication/history.js]: Can you write a comment explaining what this is for?

File: frontend/src/modules/LoginRedux.js [hunk 3]
```diff
@@ -8,8 +10,22 @@ export const IS_CHANGING_PASSWORD = "IS_CHANGING_PASSWORD";
...
+    }
+
+    if (location === PATH.login) {
+      push(PATH.dashboard);
+    }
+    return dispatch({ type: AUTHENTICATED });
+  };
+}
+
+function updateAuthenticatedState() {
```
Comment (caleybrock) [inline frontend/src/modules/LoginRedux.js]: The difference between `updateAuthenticatedState` and `authenticateAction` is confusing. I think they need re-naming and some comments.

I think this should stay as is and be used whenever you need to update the state.
`const authenticateAction = authenticated => ({ type: AUTHENTICATED, authenticated });`

Then your more complex function that handles local storage should be named something more specific like `handleAuthResponseAndState` ??

File: frontend/src/modules/LoginRedux.js (hunk 1 above)
Comment (fnormand01) [reply frontend/src/modules/LoginRedux.js]: Fixed in commit #https://github.com/code-for-canada/project-thundercat/pull/312/commits/793655aa429516a071f45af6a8d24061dd77adbe

File: frontend/src/components/authentication/history.js (hunk 2 above)
Comment (fnormand01) [reply frontend/src/components/authentication/history.js]: Added comments in commit #https://github.com/code-for-canada/project-thundercat/pull/312/commits/3176cff3c1af31353290a9afb7691050f2993c4f

File: frontend/src/modules/LoginRedux.js (hunk 3 above)
Comment (fnormand01) [reply frontend/src/modules/LoginRedux.js]: Refactored in commit #https://github.com/code-for-canada/project-thundercat/pull/312/commits/1226ba3db4c64a02aace8e13f5aef15bef0c565c
